app/pages/project/instances/instance/InstancePage.tsx

```diff
@@ -81,6 +116,8 @@ export function InstancePage() {
 
   const memory = filesize(instance.memory, { output: 'object', base: 2 })
 
+  const primaryVpcId = nics.items.length > 0 ? nics.items[0].vpcId : undefined
+
```

Comment:
It just occurred to me — I don't think we can rely on the first NIC being the primary one. We have to find the primary directly.